src/submission/validation-clinical/donor.ts

```diff
@@ -1,28 +1,119 @@
 import {
-  SubmissionValidationError,
   FieldsEnum,
   DataValidationErrors,
+  SubmittedClinicalRecord,
+  SubmissionValidationError,
+  ClinicalInfoFieldsEnum,
+  ValidatorResult,
+  ModificationType,
 } from '../submission-entities';
 import { DeepReadonly } from 'deep-freeze';
 import { Donor } from '../../clinical/clinical-entities';
 import { FileType } from '../submission-api';
+import * as utils from './utils';
 
 export const validate = async (
-  newDonorRecords: DeepReadonly<{ [clinicalType: string]: any }>,
+  newRecords: DeepReadonly<{ [clinicalType: string]: SubmittedClinicalRecord }>,
   existentDonor: DeepReadonly<Donor>,
-): Promise<any> => {
-  const errors = [];
-  const donorSubmitterId = newDonorRecords[FileType.DONOR][FieldsEnum.submitter_donor_id];
-  if (!existentDonor) {
-    errors.push({
-      type: DataValidationErrors.ID_NOT_REGISTERED,
-      fieldName: FieldsEnum.submitter_donor_id,
-      info: {
-        donorSubmitterId: donorSubmitterId,
-        value: donorSubmitterId,
-      },
-      index: newDonorRecords[FileType.DONOR].recordIndex,
-    });
+): Promise<ValidatorResult> => {
+  const errors: SubmissionValidationError[] = [];
+  const donorRecord = newRecords[FileType.DONOR];
+
+  // Preconditions: if any one of these validation failed, can't continue
+  if (!utils.checkDonorRegistered(existentDonor, donorRecord)) {
+    return {
+      type: ModificationType.ERRORSFOUND,
+      index: donorRecord.index,
+      resultArray: [
+        utils.buildSubmissionError(
+          donorRecord,
+          DataValidationErrors.ID_NOT_REGISTERED,
+          FieldsEnum.submitter_donor_id,
+        ),
+      ],
+    };
   }
-  return errors;
+
+  // cross entity donor record validation
+  checkTimeConflictWithSpecimen(existentDonor, donorRecord, newRecords[FileType.SPECIMEN], errors);
+
+  return errors.length > 0
+    ? { type: ModificationType.ERRORSFOUND, index: donorRecord.index, resultArray: errors }
+    : await calculateStats(donorRecord, existentDonor);
```

Comment:
this function name is confusing... its not actually calculating stats, just checking for new/updated/not-updated status